api/src/main/java/io/opencensus/stats/MeasureMap.java

```diff
@@ -51,6 +52,17 @@
    */
   public abstract MeasureMap put(MeasureLong measure, long value);
 
+  /**
+   * Associate the contextual information of an {@code Exemplar} to this {@link MeasureMap}.
+   *
+   * <p>If this method is called multiple times, only the last string map will be kept.
+   *
+   * @param attachments contextual information of an {@code Exemplar}.
+   * @return this
+   * @since 0.16
+   */
+  public abstract MeasureMap withAttachments(Map<String, String> attachments);
```

Comment:
It other parts of the API, we use a "with" prefix for methods that manipulate the context or return a copy of an immutable object, so I don't think this method should start with "with".  Other possibilities are "setAttachments", if the method replaces existing attachments, or "putAttachements", if the method merges new and existing attachments.

Reply:
Renamed to `putAttachements`.